src/nvim/screen.c

```diff
@@ -5017,9 +5017,9 @@ win_redr_custom (
     maxwidth = default_grid.Columns;
     use_sandbox = was_set_insecurely((char_u *)"tabline", 0);
   } else {
-    row = wp->w_height;
+    row = wp->w_grid.Rows + wp->w_grid.OffsetRow;
```

Comment:
it is probably better to use the original `wp->w_winrow+wp->w_height` here. Because when we will allow UI:s to change the "inner" grid size, the statusline shouldn't be affected. So in general (i e except when we have a reason to do otherwise), code that draws on global grid should use `wp->w_*` varables, code that draws on local grid should use `wp->w_grid.*` variables.